hr_timesheet_sheet_period/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from datetime import datetime
+from openerp.osv import fields, osv, orm
+from openerp.tools.translate import _
+
+
+class HrTimesheetSheet(orm.Model):
+    _inherit = "hr_timesheet_sheet.sheet"
+
+    _columns = {
+        'hr_period_id': fields.many2one(
+            'hr.period', string='Period',
+            readonly=True, states={
+                'draft': [('readonly', False)],
+                'new': [('readonly', False)]})
+    }
+
+    def onchange_pay_period(self, cr, uid, id, hr_period_id, context=None):
+        if hr_period_id:
+            period = self.pool['hr.period'].browse(
+                cr, uid, hr_period_id, context=context)
+            return {'value': {'date_from': period.date_start,
+                              'date_to': period.date_stop}}
+        return {}
+
+    def _get_current_pay_period(self, cr, uid, context=None):
+        period_obj = self.pool['hr.period']
+        date_today = datetime.today().strftime('%Y-%m-%d')
+        period_ids = period_obj.search(cr, uid,
+                                       [('date_start', '<=', date_today),
+                                        ('date_stop', '>=', date_today)],
+                                       context=context)
+        if period_ids:
+            return period_ids[0]
+        else:
+            return False
```

Comment:
You don't check which type of schedule the employee's contract uses ? Some employees are paid each weeks, others bi-weekly, etc. The module hr_period enables to manage multiple schedules concurrently. You should filter the periods according to the schedule type.